funnel/models/proposal.py

```diff
@@ -123,11 +127,11 @@ class Proposal(BaseScopedIdNameMixin, CoordinatesMixin, db.Model):
     state = StateManager('_state', PROPOSAL_STATE, doc="Current state of the proposal")
 
     votes_id = db.Column(db.Integer, db.ForeignKey('votespace.id'), nullable=False)
-    votes = db.relationship(VoteSpace, uselist=False,
+    votes = db.relationship(VoteSpace, uselist=False, lazy='joined',
                             cascade='all, delete-orphan', single_parent=True)
 
     comments_id = db.Column(db.Integer, db.ForeignKey('commentspace.id'), nullable=False)
-    comments = db.relationship(CommentSpace, uselist=False,
+    comments = db.relationship(CommentSpace, uselist=False, lazy='joined',
```

Comment:
These 3 lazies are to improve funnel loading time.

Reply:
Agree on votes and comments, [but you need `lazy='select'` since `'joined'` does an eager load](http://docs.sqlalchemy.org/en/latest/orm/relationship_api.html#sqlalchemy.orm.relationship.params.lazy).

Reply:
@jace `select` seems like the default behavior, where it emits a select query when needed, but for the same reason, when loading funnel front page, for rootconf locally, it's making >300 queries right now. maybe I need to move these lazies and try to do joinedload while loading the front page?